python/src/aiconfig/editor/server/server.py

```diff
@@ -162,24 +171,102 @@ def create() -> FlaskResponse:
 
 
 @app.route("/api/run", methods=["POST"])
-async def run() -> FlaskResponse:
+def run():
+    EXCLUDE_OPTIONS = {
+        "prompt_index": True,
+        "file_path": True,
+        "callback_manager": True,
+    }
     state = get_server_state(app)
     aiconfig = state.aiconfig
     request_json = request.get_json()
+    prompt_name: str = request_json.get("prompt_name", "get_activities") # Hard-coded
+    params = request_json.get("params", {}) #Fixed in https://github.com/lastmile-ai/aiconfig/pull/668
+    stream = request_json.get("stream", True)
+
+    # Define stream callback and queue object for streaming results
+    output_text_queue = QueueIterator()
+    def update_output_queue(data, _accumulated_data, _index) -> None:
```

Comment:
Is this going to work for all the model parser? I thought the idea of the callback was that specific models need specific logic

Reply:
If your model parser supports streaming, you need to implement something that returns the streaming results into a `stream_callback` object, otherwise nobody will be able to read the results as they're happening. If you don't do this in your model parser, that's on you, not us. 

If there are ever cases where you can stream non-text data formats, we'll make a note and try to follow up later